crates/orchestrator/src/jobs/mod.rs

```diff
@@ -227,18 +229,23 @@ pub async fn process_job(id: Uuid, config: Arc<Config>) -> Result<(), JobError>
 
     tracing::debug!(job_id = ?id, job_type = ?job.job_type, "Getting job handler");
     let job_handler = factory::get_job_handler(&job.job_type).await;
-    let external_id = match job_handler.process_job(config.clone(), &mut job).await {
-        Ok(external_id) => {
+    let external_id = match AssertUnwindSafe(job_handler.process_job(config.clone(), &mut job)).catch_unwind().await {
+        Ok(Ok(external_id)) => {
             tracing::debug!(job_id = ?id, "Successfully processed job");
             external_id
         }
-        Err(e) => {
+        Ok(Err(e)) => {
             // TODO: I think most of the times the errors will not be fixed automatically
             // if we just retry. But for some failures like DB issues, it might be possible
             // that retrying will work. So we can add a retry logic here to improve robustness.
             tracing::error!(job_id = ?id, error = ?e, "Failed to process job");
             return move_job_to_failed(&job, config.clone(), format!("Processing failed: {}", e)).await;
         }
+        Err(_) => {
+            tracing::error!(job_id = ?id, "Job handler panicked during processing");
+            return move_job_to_failed(&job, config.clone(), "Job handler panicked during processing".to_string())
```

Comment:
let's add panic error here

Reply:
after catching the panic? we should throw an panic error?

also as per the current logic, if a job x fails, job x+1 won't get process? if that's the case, we should panic

Reply:
no no, error message that goes inside `move_job_to_failed` should include panic error message.

> also as per the current logic, if a job x fails, job x+1 won't get process? if that's the case, we should panic

I think this will be handle by the worker is enabled check (inside run_worker iirc)

Reply:
I see got it

Reply:
added